src/framework/components/element/image-element.js

```diff
@@ -1034,13 +1043,15 @@ class ImageElement {
                 }
             }
 
-            this._materialAsset = _id;
-            if (this._materialAsset) {
-                const asset = assets.get(this._materialAsset);
+            this._materialAsset = null;
+            if (_id) {
+                const asset = assets.get(_id);
                 if (!asset) {
                     this.material = null;
+                    this._materialAsset = _id;
```

Comment:
`this._materialAsset = _id;` is in both branches of the `if`.

Can we replace only 1037 by `this._materialAsset = _id || null;`?

Reply:
The point is that if `this._materialAsset` is not `null` when we set `this.material` to `null`, `this.material` setter will call `this.materialAsset` to `null` which we don't want. So I had to make sure it's `null` before assigning to the right value after processing with `material` reset.

It's also not possible to change `this._material` directly as it won't reset the displayed material.

So yeah a bit upside-down but I couldn't find any better way to do it...

Reply:
@querielo Any further comments here or is this particular conversation resolved?

Reply:
I moved to the other solution
```
this._materialAsset = null;
this.material = null;

this._materialAsset = _id;
```

So it's more clear that the `null` assignment is only for material `null`ing and reduces the number of changed lines, as it's only to reduces call from `material` setter and avoid a loop between the two.

In fact the bug was coming from material setter not changing the `_materialAsset` and the enable callback using the `_materialAsset` if any, overriding any existing `_material` if it was from a different ID/no ID. So the main point is to remove `_materialAsset` inside `material` setter which was then making a loop with `materialAsset` setter if `_materialAsset` was non-`null` or different, why the `null`.